code/modules/atmospherics/machinery/components/trinary_devices/hypertorus.dm

```diff
@@ -0,0 +1,893 @@
+	icon_state_open = "waste_output"
+	icon_state_off = "waste_output"
+	circuit = /obj/item/circuitboard/machine/HFR_waste_output
+
+/obj/machinery/atmospherics/components/unary/hypertorus/moderator_input
+	name = "HFR moderator input port"
+	desc = "Moderator port for the Hypertorus Fusion Reactor, designed to move gases inside the machine to cool and control the flow of the reaction."
+	icon_state = "moderator_input"
+	icon_state_open = "moderator_input"
+	icon_state_off = "moderator_input"
+	circuit = /obj/item/circuitboard/machine/HFR_moderator_input
+
+/obj/machinery/hypertorus
+	name = "hypertorus_core"
+	desc = "hypertorus_core"
+	icon = 'icons/obj/atmospherics/components/hypertorus.dmi'
+	icon_state = "core"
+	move_resist = INFINITY
+	anchored = FALSE
+	density = TRUE
+	var/active = FALSE
+
+/obj/machinery/hypertorus/ComponentInitialize()
+	. = ..()
+	AddComponent(/datum/component/simple_rotation,ROTATION_ALTCLICK | ROTATION_CLOCKWISE | ROTATION_COUNTERCLOCKWISE | ROTATION_VERBS )
+
+/obj/machinery/hypertorus/wrench_act(mob/user, obj/item/I)
+	. = ..()
+	if(!active)
+		anchored = !anchored
+	else
+		message_admins("Is active")
+
+/obj/machinery/hypertorus/proc/activate()
+	return
+
+/obj/machinery/hypertorus/proc/deactivate()
+	return
+
+/obj/machinery/atmospherics/components/binary/hypertorus/core
+	name = "HFR core"
+	desc = "This is the Hypertorus Fusion Reactor core, an advanced piece of technology to finely tune the reaction inside of the machine. It has I/O for cooling gases."
+	icon = 'icons/obj/atmospherics/components/hypertorus.dmi'
+	icon_state = "core"
+	circuit = /obj/item/circuitboard/machine/HFR_core
+	pipe_flags = PIPING_ONE_PER_TURF | PIPING_DEFAULT_LAYER_ONLY
+	density = TRUE
+
+	///Checks if the machine state is active (all parts are connected)
+	var/active = FALSE
+	///Checks if the user has started the machine
+	var/fusion_started = FALSE
+	var/next_slowprocess = 0
+	///Stores the informations of the interface machine
+	var/obj/machinery/hypertorus/interface/linked_interface
+	///Stores the information of the moderator input
+	var/obj/machinery/atmospherics/components/unary/hypertorus/moderator_input/linked_moderator
+	///Stores the information of the fuel input
+	var/obj/machinery/atmospherics/components/unary/hypertorus/fuel_input/linked_input
+	///Stores the information of the waste output
+	var/obj/machinery/atmospherics/components/unary/hypertorus/waste_output/linked_output
+	///Stores the information of the corners of the machine
+	var/list/corners = list()
+	///Stores the information of the fusion gasmix
+	var/datum/gas_mixture/internal_fusion
+	///Stores the information of the buffer gasmix (used to move gases around, may be removed)
+	var/datum/gas_mixture/buffer
+	///Stores the information of the output gasmix (used to move gases around, may be removed)
+	var/datum/gas_mixture/internal_output
+	///Stores the information of the moderators gasmix
+	var/datum/gas_mixture/moderator_internal
```

Comment:
Be sure to null these vars on destruction, and to register to their early destruction if that may happen, handling it appropriately.

Reply:
yes, some of them won't be needed in the end product (looking at you buffer).